lib/Dialect/EmitC/IR/EmitCDialect.cpp

```diff
@@ -50,6 +50,18 @@ void emitc::buildTerminatedBody(OpBuilder &builder, Location loc) {
   builder.create<emitc::YieldOp>(loc);
 }
 
+//===----------------------------------------------------------------------===//
+// ApplyOp
+//===----------------------------------------------------------------------===//
+
+static LogicalResult verify(mlir::emitc::ApplyOp op) {
+  // Callee must not be empty.
+  if (op.applicableOperator().empty()) {
+    return op.emitOpError("applicable operator must not be empty");
+  }
```

Comment:
Only allow "&" and "*" for now. We can then verify types between operand and result in a follow up (we might need a ptr type).

Reply:
Good point! Extended the verifier.